File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int i = 1;
		
		while(true) {
			if(i*(i+1)/2 > n) {
				break;
			}
			
			if(i % 2 == 1 && n % i == 0) {
				System.out.println(i);
				answer++;
			} else if(i % 2 == 0 && n % i == i/2){
				System.out.println(i);
				answer++;
			}
			
			i++;
		}
        return answer;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    let answer = 0;
    for (let i = 1; i*(i+1)/2<=n; i++){
        if (i % 2 === 1 && n % i === 0) answer++;
        if (i % 2 === 0 && n % i === i / 2) answer++;
    }
    
    
    return answer;
}